transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -461,40 +552,75 @@ mod rct_bulletproofs_tests {
                 real_input_indices,
                 input_secrets,
                 output_values_and_blindings,
+                block_version,
+                token_id,
             }
         }
 
         fn get_output_commitments(&self) -> Vec<CompressedCommitment> {
             self.output_values_and_blindings
                 .iter()
-                .map(|(value, blinding)| CompressedCommitment::new(*value, *blinding))
+                .map(|(value, blinding)| {
+                    CompressedCommitment::new(*value, *blinding, &self.generator())
+                })
                 .collect()
         }
+
+        fn sign<RNG: RngCore + CryptoRng>(
+            &self,
+            fee: u64,
+            rng: &mut RNG,
+        ) -> Result<SignatureRctBulletproofs, Error> {
+            SignatureRctBulletproofs::sign(
+                self.block_version,
+                &self.message,
+                &self.rings,
+                &self.real_input_indices,
+                &self.input_secrets,
+                &self.output_values_and_blindings,
+                fee,
+                self.token_id,
+                rng,
+            )
+        }
+
+        fn sign_without_balance_check<RNG: RngCore + CryptoRng>(
+            &self,
+            fee: u64,
+            rng: &mut RNG,
+        ) -> Result<SignatureRctBulletproofs, Error> {
+            sign_with_balance_check(
+                self.block_version,
+                &self.message,
+                &self.rings,
+                &self.real_input_indices,
+                &self.input_secrets,
+                &self.output_values_and_blindings,
+                fee,
+                self.token_id,
+                false,
+                rng,
+            )
+        }
     }
 
     proptest! {
-        #![proptest_config(ProptestConfig::with_cases(3))]
+        #![proptest_config(ProptestConfig::with_cases(6))]
 
         #[test]
         // `sign`should return an error if `rings` is empty.
         fn sign_rejects_empty_rings(
             num_inputs in 1..8usize,
             num_mixins in 1..17usize,
             seed in any::<[u8; 32]>(),
+            block_version in 2..=3u32,
         ) {
+            let block_version: BlockVersion = block_version.try_into().unwrap();
             let mut rng: StdRng = SeedableRng::from_seed(seed);
-            let mut params = SignatureParams::random(num_inputs, num_mixins, &mut rng);
+            let mut params = SignatureParams::random(block_version,num_inputs, num_mixins, &mut rng);
```

Comment:
```suggestion
            let mut params = SignatureParams::random(block_version, num_inputs, num_mixins, &mut rng);
```

Reply:
i wonder why cargo fmt isn't doing this

Reply:
I think because it's inside a macro call